draft-sharabayko-srt.md

```diff
@@ -41,9 +46,159 @@ TODO Introduction
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+{: #datapacket title="data packet structure"}
+
+F (1 bit): 
+: Packet Flag. The flag value of data packet must be 0.
+
+Packet Sequence Number (31 bits):
+: 
+
+PP (2 bits):
+: Position of Packet. This field indicates the position of packet.
+  The value "10b" means the first packet, "00b" is packets in the middle, "01b" is the last packet.
+  If it is a single data packet, the value is "11b".
+
+O (1 bit):
+: Order.  Indicates whether the message should be delivered in order or not.
+  This value is not used in Live Transmission Mode({{live-transmission-mode}}).
+
+Enc (2 bits):
+: Encryption Flag. The flag bits indicate whether or not data is encrypted.
+  The value "00b" is not encrypted, "01b" means ecnrypted data with even key, 
+ "11b" signifies ecrypted data with odd key.
+
+R (1 bit):
+: Retransmitted Packet. This flag bit is clear when a packet is transmitted very first time.
+  If a packet is retransmitted, the flag is set as "1".
+
+Message Number (26 bits):
+: 
+
+Timestamp (32 bits):
+: The time stamp when the packet is sent. 
+  This is relative time value starting from when the connection is established.
+
+Destination Socket ID (32 bits):
+: 
+
+Data (variable length):
+:
+
+# Control Packets
+
+If the flag bit of SRT packet is set as "1", it should have a following structure as a control packet.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|F|         Message Type        |            Reserved           |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                   Type-specific Information                   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           Timestamp                           |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                     Destination Socket ID                     |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                   Control Information Field                   +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+{: #controlpacket title="control packet structure"}
+
+F (1 bit): 
+: Packet Flag. The control packet must set this flag as "1".
+
+Message Type (15 bits):
+: Control Message Type. The use of these bits is determined
+  by the control message type definition.
+
+  - "0": Handshake
+  - "1": Keep Alive
+  - "2": ACK
+  - "3": NACK (Loss Report)
+  - "4": Congestion Warning
+  - "5": Shutdown
+  - "6": ACKACK
+  - "7": Drop Request
+  - "8": Peer Error
```

Comment:
No information about *Drop Request* and *Peer Error*.

Reply:
Good point. We can create an issue for this so that it does not block us.